src/test/java/co/cask/cdap/guides/PageRankAppTest.java

```diff
@@ -49,40 +49,33 @@ public void test() throws Exception {
     ApplicationManager appManager = deployApplication(PageRankApp.class);
 
     // Send a stream events to the Stream
-    StreamWriter streamWriter = appManager.getStreamWriter(PageRankApp.PAGE_RANK_BACKLINK_STREAM);
-    streamWriter.send(URL_PAIR1);
-    streamWriter.send(URL_PAIR2);
-    streamWriter.send(URL_PAIR3);
-    streamWriter.send(URL_PAIR4);
-    streamWriter.send(URL_PAIR5);
-    streamWriter.send(URL_PAIR6);
-
+    StreamManager streamManager = getStreamManager(PageRankApp.PAGE_RANK_BACKLINK_STREAM);
+    streamManager.send(URL_PAIR1);
+    streamManager.send(URL_PAIR2);
+    streamManager.send(URL_PAIR3);
+    streamManager.send(URL_PAIR4);
+    streamManager.send(URL_PAIR5);
+    streamManager.send(URL_PAIR6);
 
     // Start the Spark Program
-    SparkManager sparkManager = appManager.startSpark(PageRankSpark.class.getSimpleName());
+    SparkManager sparkManager = appManager.getSparkManager(PageRankSpark.class.getSimpleName()).start();
     sparkManager.waitForFinish(60, TimeUnit.SECONDS);
 
-    ServiceManager serviceManager = appManager.startService(PageRankApp.PAGE_RANK_RANKS_SERVICE);
+    ServiceManager serviceManager = appManager.getServiceManager(PageRankApp.PAGE_RANK_RANKS_SERVICE).start();
     serviceManager.waitForStatus(true);
 
-    String response = requestService(new URL(serviceManager.getServiceURL(), PageRankHandler.PAGE_RANKS_RANK_HANDLER +
-      "?url=http://example.com/page1"));
+    //Query for rank and verify it
+    URL totalHitsURL = new URL(serviceManager.getServiceURL(15, TimeUnit.SECONDS),
+                               PageRankHandler.PAGE_RANKS_RANK_HANDLER);
 
-    //pagerank in any case should be more than 0.0
-    Assert.assertTrue(Double.parseDouble(response) > 0.0);
+    HttpResponse response = HttpRequests.execute(HttpRequest.post(totalHitsURL)
```

Comment:
It makes sense to have retries in integration tests but I don't think we have to worry about it in unit tests.